src/clib/lib/enkf/ecl_config.cpp

```diff
@@ -1,483 +0,0 @@
-                       ui_return_get_last_error(ui_return));
-        ui_return_free(ui_return);
-    }
-}
-
-static void handle_has_data_file_key(ecl_config_type *ecl_config,
-                                     const config_content_type *config) {
-    const char *data_file =
-        config_content_get_value_as_abspath(config, DATA_FILE_KEY);
-    ui_return_type *ui_return =
-        ecl_config_validate_data_file(ecl_config, data_file);
-    if (ui_return_get_status(ui_return) == UI_RETURN_OK)
-        ecl_config_set_data_file(ecl_config, data_file);
-    else
-        util_abort("%s: problem setting ECLIPSE data file (%s)\n", __func__,
-                   ui_return_get_last_error(ui_return));
-    ui_return_free(ui_return);
-}
-
-static void handle_has_grid_key(ecl_config_type *ecl_config,
-                                const config_content_type *config) {
-    const char *grid_file =
-        config_content_get_value_as_abspath(config, GRID_KEY);
-
-    ui_return_type *ui_return = ecl_config_validate_grid(ecl_config, grid_file);
-    if (ui_return_get_status(ui_return) == UI_RETURN_OK)
-        ecl_config_set_grid(ecl_config, grid_file);
-    else
-        util_abort("%s: failed to set grid file:%s  Error:%s \n", __func__,
-                   grid_file, ui_return_get_last_error(ui_return));
-
-    ui_return_free(ui_return);
-}
-
-static void handle_has_refcase_key(ecl_config_type *ecl_config,
-                                   const config_content_type *config) {
-    const char *refcase_path =
-        config_content_get_value_as_abspath(config, REFCASE_KEY);
-
-    if (!ecl_config_load_refcase(ecl_config, refcase_path))
-        fprintf(stderr, "** Warning: loading refcase:%s failed \n",
-                refcase_path);
-}
-
-static void
-handle_has_schedule_prediction_file_key(ecl_config_type *ecl_config,
-                                        const config_content_type *config) {
-    const config_content_item_type *pred_item =
-        config_content_get_item(config, SCHEDULE_PREDICTION_FILE_KEY);
-
-    config_content_node_type *pred_node =
-        config_content_item_get_last_node(pred_item);
-    const char *template_file = config_content_node_iget_as_path(pred_node, 0);
-    ecl_config_set_schedule_prediction_file(ecl_config, template_file);
-}
-
-void ecl_config_init(ecl_config_type *ecl_config,
-                     const config_content_type *config) {
-    if (config_content_has_item(config, ECLBASE_KEY))
-        handle_has_eclbase_key(ecl_config, config);
-
-    if (config_content_has_item(config, DATA_FILE_KEY))
-        handle_has_data_file_key(ecl_config, config);
-
-    if (config_content_has_item(config, GRID_KEY))
-        handle_has_grid_key(ecl_config, config);
-
-    if (config_content_has_item(config, REFCASE_KEY))
-        handle_has_refcase_key(ecl_config, config);
-
-    if (ecl_config->can_restart)
-        fprintf(
-            stderr,
```

Comment:
We should probably have similar warning message printed, and test that we get the expected warning: pytest.warns(UserWarning, match="..."):.